docs/load_ldoc.cmake

```diff
@@ -13,8 +13,11 @@ foreach(doc_file_name ${doc_files})
     # Remove the file extension
     string(REGEX REPLACE "\\.ldoc" "" DOC_FILE_NAME ${doc_file_name})
 
+    # There is a trailing \n, remove it or it cannot be included in existing blocks
+    string(REGEX REPLACE "\n$" "" doc_file_content "${doc_file_content}")
+
     # Create a new variable usable from lua files
-    set(DOC_${DOC_FILE_NAME}_COMMON "Imported documentation\n\n${doc_file_content}")
+    set(DOC_${DOC_FILE_NAME}_COMMON "${doc_file_content}")
```

Comment:
This should cause lots of changes in the docs. Is this really okay? (I have honestly no idea)
At least the commit message could mention/explain this... (and also why it is needed)

Reply:
Its necessary to be able to do

```
--- Foobar.
-- @tparam table args
-- @tparam args.new_property
-- @DOC_wibox_constructor_COMMON@
-- @treturn boolean
function new(args) end
```

Without this, the previous result was

```
--- Foobar.
-- @tparam table args
-- @tparam args.new_property

-- Imported documentation

-- @tparam args.shared_property

-- @treturn boolean
function new(args) end
```

instead of :

```
--- Foobar.
-- @tparam table args
-- @tparam args.new_property
-- @tparam args.shared_property
-- @treturn boolean
function new(args) end
```

It doesn't change anything for existing doc beside the lack of newlines as long as there is a new line between the macros

Ok:

```
--@DOC_widget_COMMON@

--@DOC_object_COMMON@
```

Potentially not ok, but nevertheless mostly ok:

```
--@DOC_widget_COMMON@
--@DOC_object_COMMON@
```